packages/main/src/plugin/temp-file-service.spec.ts

```diff
@@ -0,0 +1,288 @@
+/**********************************************************************
+ * Copyright (C) 2025 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import * as fs from 'node:fs';
+import * as os from 'node:os';
+import * as path from 'node:path';
+
+import { beforeEach, describe, expect, test, vi } from 'vitest';
+
+import { TempFileService } from './temp-file-service.js';
+
+vi.mock('node:fs');
+vi.mock('node:os');
+vi.mock('node:path');
+
+const mockWriteFile = vi.fn();
+const mockUnlink = vi.fn();
+const mockTmpDir = vi.fn().mockReturnValue('/tmp');
+const mockJoin = vi.fn().mockImplementation((...args) => args.join('/'));
+
+vi.mocked(fs.promises).writeFile = mockWriteFile;
+vi.mocked(fs.promises).unlink = mockUnlink;
+vi.mocked(os).tmpdir = mockTmpDir;
+vi.mocked(path).join = mockJoin;
+
+describe('TempFileService', () => {
```

Comment:
You don't need a describe at the root, as the filename is enough, let's avoid unnecessary indentation